src/mesh/mesh.cc

```diff
@@ -1118,35 +1102,36 @@ vector<vector<unsigned int> > const & Mesh::node_elements() {
 }
 
 
-void Mesh::init_element_vector(unsigned int size) {
+void MeshBase::init_element_vector(unsigned int size) {
 	element_vec_.clear();
     element_ids_.clear();
 	element_vec_.reserve(size);
     element_ids_.reserve(size);
-	bc_element_tmp_.clear();
-	bc_element_tmp_.reserve(size);
-	bulk_size_ = 0;
-	boundary_loaded_size_ = 0;
 }
 
 
-void Mesh::init_node_vector(unsigned int size) {
+void MeshBase::init_node_vector(unsigned int size) {
 	nodes_.reinit(size);
 	node_ids_.clear();
 	node_ids_.reserve(size);
 }
 
 
-Element * Mesh::add_element_to_vector(int id) {
-    element_vec_.push_back( Element() );
-    Element * elem = &element_vec_.back(); //[element_vec_.size()-1];
-    element_ids_.add_item((unsigned int)(id));
+Element * MeshBase::add_element_to_vector(int id, bool is_boundary) {
```

Comment:
The distinction should rather be done in GMSH reader and one should rather use mesh->add_element_to_vector or bc_mesh->add_element_to_vector.